### PR TITLE
Fix: issues affecting the code quality

### PR DESCRIPTION
**Pull request by withshubh**
#### Description
This PR fixes a few issues that were affecting the code quality.

#### Summary of changes

- Prefer rescuing `StandardError` over `Exception`
- Prefix any unused method arguments with an underscore
- Remove redundant and unnecessary require statement
- Replaces the old OpenSSL algorithmic constants with the newer strings initializers
- Remove redundant string coercion
- Remove useless access modifiers

### CODE REVIEW

File: lib/onelogin/ruby-saml/response.rb [hunk 1]
```diff
@@ -727,7 +727,7 @@ def validate_issuer
       # @return [Boolean] True if the SessionNotOnOrAfter of the AuthnStatement is valid, otherwise (when expired) False if soft=True
       # @raise [ValidationError] if soft == false and validation fails
       #
-      def validate_session_expiration(soft = true)
```
Comment (pitbulk) [inline lib/onelogin/ruby-saml/response.rb:730]: I believe we can simply remove the param

Comment (withshubh) [reply lib/onelogin/ruby-saml/response.rb:730]: @pitbulk removed!:sparkles:

**Comment by withshubh** [comment]
Hi @pitbulk :wave: 

Please review the PR :sparkles:

File: lib/onelogin/ruby-saml/response.rb [hunk 2]
```diff
@@ -727,7 +727,7 @@ def validate_issuer
...
+      def validate_session_expiration()
```
Comment (johnnyshields) [inline lib/onelogin/ruby-saml/response.rb:730]: empty parenthesis not needed here

**Comment by johnnyshields** [comment]
👍 looks good to merge aside from small comment above

File: lib/onelogin/ruby-saml/response.rb (hunk 2 above)
Comment (withshubh) [inline lib/onelogin/ruby-saml/response.rb:730]:
```suggestion
      def validate_session_expiration
```

**Comment by johnnyshields** [comment]
@pitbulk let's merge this one?

**Review by pitbulk — APPROVED** [review]

**Pull request merged**